gcp/master.tf

```diff
@@ -0,0 +1,270 @@
+    port = "5050"
+  }
+}
+
+# Used for the external load balancer. The external load balancer only supports google_compute_http_health_check resource.
+resource "google_compute_http_health_check" "master-adminrouter-healthcheck" {
+  name                = "${data.template_file.cluster-name.rendered}-external-mesos-http-master-healthcheck"
+  check_interval_sec  = 30
+  timeout_sec         = 5
+  healthy_threshold   = 2
+  unhealthy_threshold = 2
+  port                = "80"
+
+  # depends_on = ["google_compute_target_pool.master-pool"]
+}
+
+ 
+
+# Provide tested AMI and user from listed region startup commands
+module "dcos-tested-gcp-oses" {
+   source   = "./modules/dcos-tested-gcp-oses"
+   os       = "${var.os}"
+}
+
+resource "google_compute_instance_group" "master" {
+  name        = "${data.template_file.cluster-name.rendered}-master-cluster"
+  description = "DC/OS Master Instance Group"
+
+  instances = [
+    "${google_compute_instance.master.*.self_link}"
+  ]
+
+  named_port {
+    name = "http"
+    port = "80"
+  }
+
+  named_port {
+    name = "https"
+    port = "443"
+  }
+
+  named_port {
+    name = "mesos-master"
+    port = "5050"
+  }
+
+  named_port {
+    name = "marathon"
+    port = "8080"
+  }
+
+  named_port {
+    name = "zookeeper"
+    port = "2181"
+  }
+
+  named_port {
+    name = "exhibitor"
+    port = "8181"
+  }
+
+  named_port {
+    name = "ssh"
+    port = "22"
+  }
+
+  zone = "${data.google_compute_zones.available.names[0]}"
+}
+
+# deploy image
+resource "google_compute_instance" "master" {
+   name         = "${data.template_file.cluster-name.rendered}-master-${count.index + 1}"
+   machine_type = "${var.gcp_master_instance_type}"
+   zone         = "${data.google_compute_zones.available.names[0]}"
+   count        = "${var.num_of_masters}"
+ 
+  labels {
+   owner = "${coalesce(var.owner, data.external.whoami.result["owner"])}"
+   expiration = "${var.expiration}"
+   name = "${data.template_file.cluster-name.rendered}-master-${count.index + 1}"
+   cluster = "${data.template_file.cluster-name.rendered}"
+  }
+
+  boot_disk {
+    initialize_params {
+      image = "${module.dcos-tested-gcp-oses.gcp_image_family}/${module.dcos-tested-gcp-oses.gcp_image_name}"
+      size  = "${var.instance_disk_size}"
+    }
+  }
+
+  connection {
+    # host = "${element(google_compute_instance.master.*.network_interface.0.access_config.0.assigned_nat_ip, count.index)}"
+    user = "${coalesce(var.gce_ssh_user, module.dcos-tested-gcp-oses.user)}"
+  }
+
+  network_interface {
+    subnetwork = "${google_compute_subnetwork.public.name}"
+    access_config {
+        // IP
+    }
+  } 
+
+  metadata {
+    sshKeys = "${coalesce(var.gce_ssh_user, module.dcos-tested-gcp-oses.user)}:${file(var.gce_ssh_pub_key_file)}"
+  }
+
+  # network_interface {
+  #   network = "default"
+  # }
+
+  # OS init script
+  provisioner "file" {
+   content = "${module.dcos-tested-gcp-oses.os-setup}"
+   destination = "/tmp/os-setup.sh"
```

Comment:
same here, /tmp vs /opt cc @branden

Reply:
Since this is the os-setup.sh which only lives and runs once, if that machine were to be ever rebooted, the files will be removed as desired since it resided in the /tmp dir.